apps/cowswap-frontend/src/modules/permit/utils/generatePermitHook.ts

```diff
@@ -138,3 +138,59 @@ async function calculateGasLimit(
     return DEFAULT_PERMIT_GAS_LIMIT
   }
 }
+
+function getCacheKey(params: PermitHookParams): string {
+  const { inputToken, chainId, account } = params
+
+  return `${CACHE_PREFIX}${inputToken.address.toLowerCase()}-${chainId}${account ? `-${account.toLowerCase()}` : ''}`
+}
+
+type CachedData = PermitHookData & {
+  nonce: number | undefined
+}
+
+/**
+ * Stores data based on `key` and `nonce`
+ */
+function save(key: string, nonce: number | undefined, data: PermitHookData): void {
```

Comment:
why did you go for storing/retrieving from local storage instead of using a persisted joitai?